packages/amplify-graphql-api-construct-tests/src/__tests__/sql-models-common.ts

```diff
@@ -19,77 +21,696 @@ export const testGraphQLAPI = async (options: {
+        getTodo(id: $id) {
+          id
+          description
+        }
+      }
+    `;
+    const getInput = {
+      id,
+    };
+
+    const getResult: any = await appSyncClient.query({
+      query: gql(getQuery),
+      fetchPolicy: 'no-cache',
+      variables: getInput,
+    });
+
+    return getResult;
+  };
+
+  const updateTodo = async (id: string, description: string): Promise<Record<string, any>> => {
+    const updateMutation = /* GraphQL */ `
+      mutation UpdateTodo($input: UpdateTodoInput!, $condition: ModelTodoConditionInput) {
+        updateTodo(input: $input, condition: $condition) {
+          id
+          description
+        }
+      }
+    `;
+    const updateInput = {
+      input: {
+        id,
+        description,
+      },
+    };
+
+    const updateResult: any = await appSyncClient.mutate({
+      mutation: gql(updateMutation),
+      fetchPolicy: 'no-cache',
+      variables: updateInput,
+    });
+
+    return updateResult;
+  };
 
-  const todo = result.body.data.createTodo;
-  expect(todo).toBeDefined();
-  expect(todo.id).toBeDefined();
-  expect(todo.description).toEqual(description);
-
-  const listResult = await graphql(
-    apiEndpoint,
-    apiKey,
-    /* GraphQL */ `
-      query LIST_TODOS {
-        listTodos {
+  const deleteTodo = async (id: string): Promise<Record<string, any>> => {
+    const deleteMutation = /* GraphQL */ `
+      mutation DeleteTodo($input: DeleteTodoInput!, $condition: ModelTodoConditionInput) {
+        deleteTodo(input: $input, condition: $condition) {
+          id
+          description
+        }
+      }
+    `;
+    const deleteInput = {
+      input: {
+        id,
+      },
+    };
+
+    const deleteResult: any = await appSyncClient.mutate({
+      mutation: gql(deleteMutation),
+      fetchPolicy: 'no-cache',
+      variables: deleteInput,
+    });
+
+    return deleteResult;
+  };
+
+  const listTodo = async (limit = 100, nextToken: string | null = null, filter: any = null): Promise<Record<string, any>> => {
+    const listQuery = /* GraphQL */ `
+      query ListTodos($limit: Int, $nextToken: String, $filter: ModelTodoFilterInput) {
+        listTodos(limit: $limit, nextToken: $nextToken, filter: $filter) {
           items {
             id
             description
           }
+          nextToken
         }
       }
-    `,
+    `;
+
+    const listResult: any = await appSyncClient.query({
+      query: gql(listQuery),
+      fetchPolicy: 'no-cache',
+      variables: {
+        limit,
+        nextToken,
+        filter,
+      },
+    });
+
+    return listResult;
+  };
+
+  const checkGenericError = async (errorMessage?: string): Promise<void> => {
+    expect(errorMessage).toBeDefined();
+    expect(errorMessage).toEqual('GraphQL error: Error processing the request. Check the logs for more details.');
+  };
+
+  // Create Todo Mutation
+  const createTodo1 = await createTodo('Todo #1');
+  const createTodo2 = await createTodo('Todo #2');
```

Comment:
What is the purpose of the detailed assertions on the second Todo? I don't see that we're doing anything to compare them, or set relationships. Are we testing something meaningful with it that we aren't getting from Todo 1? (It makes sense to create it for the `list` assertion below, I'm questioning why we have duplicate assertions for Todo1 & Todo2 for every CRUD operation). Could you gain as much confidence in the code by asserting create/get/update for a single todo, and only then creating a second todo for the list tests?